pkg/services/dashboards/database/database.go

```diff
@@ -852,14 +852,17 @@ func (d *dashboardStore) GetDashboard(ctx context.Context, query *dashboards.Get
 		}
 
 		dashboard := dashboards.Dashboard{OrgID: query.OrgID, ID: query.ID, UID: query.UID}
+		mustCols := []string{}
 		if query.Title != nil {
 			dashboard.Title = *query.Title
+			mustCols = append(mustCols, "title")
```

Comment:
Not completely familiar with MustCols yet but seems to me it would make sense to require that both `title` and `folder_id` be set if `query.Title != nil`. Same comment for if `query.FolderID != nil`

Reply:
Actually I suppose we can rely on the check on line 850 to ensure that we do include both "title" and "folder_id" in case either of them is non-nil.

Reply:
I would love to have this query re-written without the `xorm` magic

Reply:
Rn, you can say “give me a dashboard with this ID and this title”, which has the same uniqueness as title + folder. We don’t need to do it like that, but that’s why those are separated. Note that as long as orgID is properly set (…out of scope for this fix) at least one unique index will be hit, but there’s nothing preventing you from adding further constraints (give me dashboard 5 but only if it’s in folder 10)